storage.go

```diff
@@ -8,6 +8,7 @@ type Storage interface {
 	Create(name string) (URIWriteCloser, error)
 	Open(name string) (URIReadCloser, error)
 	Save(name string) (URIWriteCloser, error)
+	Append(name string) (URIWriteCloser, error)
```

Comment:
Making this change into an existing interface is a breaking change and we cannot do it (unless a v3 is made).
Additional features added on top of the initial storage API will have to be optional.

Reply:
Okay, fair, didn't think about how people who implemented the interface themselves will lose it like this. In that case though, how do we make this addition optional while allowing for `app.Storage().Append` or other convenient usage of the new API?

Reply:
You can create a new interface like Appender or StorageAppender that defines the single function. And then the builtin one implements both Storage and Appender

Reply:
for now I deleted this commit, as there's no real reason to change the storage interface at all, I'll think about how to deal with the `internal.Docs` questions that came up for me here but for now I can resolve this at least